website/docs/r/workspace.html.markdown

```diff
@@ -68,6 +68,10 @@ The following arguments are supported:
 * `assessments_enabled` - (Optional) Whether to regularly run health assessments such as drift detection on the workspace. Defaults to `false`.
 * `auto_apply` - (Optional) Whether to automatically apply changes when a Terraform plan is successful. Defaults to `false`.
 * `auto_apply_run_trigger` - (Optional) Whether to automatically apply changes for runs that were created by run triggers from another workspace. Defaults to `false`.
+* `auto_destroy_at` - (Optional) A future date/time string at which point all resources in a workspace will be scheduled for deletion. Must be a string in RFC3339 format (e.g. "2100-01-01T00:00:00Z").
+
+~> **NOTE:** `auto_destroy_at` is not intended for workspaces containing production resources. Additionally, the field is partially managed by HCP Terraform so usage with any long-lived workspaces may cause a perpetual diff.
```

Comment:
"perpetual diff" is sloppy phrasing on my part, because the HCP Terraform API will respect the value set by the provider. but there will be contention because both pieces are effectively managing the workspace. I agree with your other comment that this doesn't make as much sense in an IaC context, however we have discussed this with enterprise teams and they have expressed a desire to control this specifically through the provider.

 yeah, HCP Terraform will unset `auto_destroy_at` after a destroy run occurs. 

I did initially have this as a computed attribute but allowing the user to set "meaningful null" values added significantly more complexity (specifically where the user has set the value but then removes `auto_destroy_at` from the TF config). what's in this PR ended up feeling like the provider SDK's preferred path for allowing for meaningful null values.

Reply:
Is it the intention that a tfe_workspace without a auto_destroy_at config should unset any drifted autodestroy setting?

Usually this kind of drift correction is good but in the case of auto_destroy, it seems like terraform wouldn't be the best source of that config.

Methodology:
1. create workspace using basic config with no auto destroy
2. set up auto destroy in the UI
3. terraform apply

Outcome:
terraform plans to unset the auto destroy setting

![Screenshot 2024-05-21 at 2 14 26 PM](https://github.com/hashicorp/go-tfe/assets/174332/d9c7cb6d-a437-4149-a372-90b267275892)

Reply:
> Is it the intention that a tfe_workspace without a auto_destroy_at config should unset any drifted autodestroy setting?

By default, yes. Talked this through with @jondavidjohn and the way we're leaning (Jon correct me if I'm wrong) is how it's currently implemented because it allows the most flexibility for users. 

@brandonc your argument makes sense: there are cases where the user may want to ignore changes to this specific field. If that's the case we can recommend that they use an `ignore_changes` here. My argument is that as a user, there are cases where I would explicitly want to be alerted if somebody set auto destroy on my workspace – in this case I always want to show the diff. 

If we specifically ignore the diff within the provider there's no way to override that behavior as a user, but there is more flexibility if we default to showing a diff.